internal/providers/packet/packet.go

```diff
@@ -48,3 +62,107 @@ func FetchConfig(f resource.Fetcher) (types.Config, report.Report, error) {
 
 	return util.ParseConfig(f.Logger, data)
 }
+
+// FetchStatus is invoked by oem.Status() when igintion fails
+// POSTS if s==fail, a message to Packet's provision setup Timeline
+func FetchStatus(f resource.Fetcher, s string) error {
+
+	type Message struct {
+		Message string `json:"message"`
+	}
+
+	// fetch JSON from https://metadata.packet.net/metadata
+	data, err := f.FetchToBuffer(MetadataUrl, resource.FetchOptions{
+		Headers: nil,
+	})
+	if data == nil {
+		log.Fatalf("data is nil")
+	}
+
+	if err != nil {
+		fmt.Println("Error: ", err)
+	}
+
+	type Metadata struct {
```

Comment:
nit:
Rather than declaring this type, which is only used once inline, can you not just do this when assinging the `metadata` variable:

```golang
metadata := struct {
    PhoneHomeURL string `json:"phone_home_url"`
}{}
```

If you do want to keep the type I think it should be moved outside of the function.